openml/extensions/sklearn/extension.py

```diff
@@ -1573,6 +1577,16 @@ def _prediction_to_probabilities(y: np.ndarray, classes: List[Any]) -> np.ndarra
             else:
                 model_classes = used_estimator.classes_
 
+            if not isinstance(model_classes, list):
+                model_classes = model_classes.tolist()
+
+            # to handle the case when dataset is numpy and categories are encoded
+            # however the class labels stored in task are still categories
+            if isinstance(y_train, np.ndarray) and \
+                    isinstance(typing.cast(List[str], task.class_labels)[0], str):
```

Comment:
I believe this should only be `typing.cast[List]` as we don't know whether it's a string, right?